apps/login/app/(login)/register/page.tsx

```diff
@@ -3,24 +3,46 @@ import {
   getPasswordComplexitySettings,
   server,
 } from "#/lib/zitadel";
-import RegisterForm from "#/ui/RegisterForm";
+import RegisterFormWithoutPassword from "#/ui/RegisterFormWithoutPassword";
+import SetPasswordForm from "#/ui/SetPasswordForm";
+
+export default async function Page({
+  searchParams,
+}: {
+  searchParams: Record<string | number | symbol, string | undefined>;
+}) {
+  const { firstname, lastname, email } = searchParams;
+
+  const setPassword = !!(firstname && lastname && email);
 
-export default async function Page() {
   const legal = await getLegalAndSupportSettings(server);
   const passwordComplexitySettings = await getPasswordComplexitySettings(
     server
   );
 
-  return (
+  return setPassword ? (
+    <div className="flex flex-col items-center space-y-4">
+      <h1>Set Password</h1>
+      <p className="ztdl-p">Set the password for your account</p>
+
+      {legal && passwordComplexitySettings && (
```

Comment:
Why do we need legal && passwordComplexitySettings to render the SetPasswordForm?
Can't we fall back to defaults?

Like this, we just render nothing if one of these is falsy, correct?

Reply:
This is business logic. We can decide to have a fallback for the complexity settings, but legal is required to have urls for the privacy policy and ToS.. I decided to throw an error if one of the commands fails.

Reply:
if one of both are missing, an error is thrown and the error boundary is rendered.